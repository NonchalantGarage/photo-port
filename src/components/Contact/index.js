import React, { useState } from "react";
import { validateEmail } from "../../utils/helpers";

function ContactForm() {
  const [formState, setFormState] = useState({
    name: "",
    email: "",
    message: "",
  });
  const [errorMessage, setErrorMessage] = useState("");

  function handleSubmit(event) {
    event.preventDefault();
    console.log(formState);
  }

  function handleChange(event) {
    if (event.target.name === "email") {
      const isValid = validateEmail(event.target.value);
      console.log(isValid);
      // isValid conditional statement
      if (!isValid) {
        setErrorMessage("Your email is invalid");
      } else {
        if (!event.target.value.length) {
          setErrorMessage(`${event.target.name} is required`);
        } else {
          setErrorMessage("");
        }

        if (!errorMessage) {
          setFormState({
            ...formState,
            [event.target.name]: event.target.value,
          });
        }
      }
    }
    console.log("errorMessage", errorMessage);
  }

  return (
    <section>
      <h1>Contact Me</h1>
      <form id="contact-from" onSubmit={handleSubmit}>
        <div>
          <label htmlFor="name">Name:</label>
          <input
            type="text"
            name="name"
            defaultValue={formState.name}
            onBlur={handleChange}
          ></input>
        </div>
        <div>
          <label htmlFor="email">Email:</label>
          <input
            type="email"
            name="email"
            defaultValue={formState.email}
            onBlur={handleChange}
          ></input>
        </div>
        <div>
          <label htmlFor="message">Message:</label>
          <textarea
            type="message"
            rows="5"
            defaultValue={formState.message}
            onBlur={handleChange}
          />
        </div>
        {errorMessage && (
          <div>
            <p className="error-text">{errorMessage}</p>
          </div>
        )}
        <button data-testid= "form" type="submit">Submit</button>
      </form>
    </section>
  );
}

export default ContactForm;
