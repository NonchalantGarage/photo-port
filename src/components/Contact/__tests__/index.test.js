import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Contact from "..";

afterEach(cleanup);

describe("Contact component", () => {
  // baseline test
  it("renders", () => {
    render(<Contact />);
  });

  it("matches snapshot", () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("inserts text into the links", () => {
    // Arrange
    const { getByTestId } = render(<Contact />);
    // Assert
    expect(getByTestId("form")).toHaveTextContent("Submit");
  });
});
