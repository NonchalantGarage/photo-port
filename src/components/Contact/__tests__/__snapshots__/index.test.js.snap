// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h1>
      Contact Me
    </h1>
    <form
      id="contact-from"
    >
      <div>
        <label
          for="name"
        >
          Name:
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="email"
        >
          Email:
        </label>
        <input
          name="email"
          type="email"
          value=""
        />
      </div>
      <div>
        <label
          for="message"
        >
          Message:
        </label>
        <textarea
          rows="5"
          type="message"
        />
      </div>
      <button
        data-testid="form"
        type="submit"
      >
        Submit
      </button>
    </form>
  </section>
</DocumentFragment>
`;
